Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+                    answer = questdlg(sprintf('%s\nDo you want to override the other client?', exception),...
+                        mfilename, 'Yes', 'No', 'No');
+                    if strcmp(answer,'Yes')
+                        % Override and recall
+                        obj.com('force_client');
+                        reply = obj.hwserver.com(obj.moduleName, fn, varargin{:});
+                        return
+                    else
+                        error(exception);
+                    end
+                end
+                rethrow(err);
+            end
+        end
+        
+        function callGetWavelength(obj)             % Calls the solstis method 'get_wavelength' and fills prefs in.
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver) || isempty(obj.moduleName)
+                obj.status = obj.no_server;
+                obj.tuning = false;
+%                 obj.wavelength = NaN;
+                obj.wavelength_lock = NaN;
+            else
+                reply = obj.com('get_wavelength', 'solstis');
+                
+                obj.status = obj.statusList{reply.status+1};
+                obj.tuning = reply.status == 2;
+%                 obj.wavelength = reply.current_wavelength;    % We will preference the wavemeter.
+                obj.wavelength_lock = logical(reply.lock_status);
+            end
+        end
+        function callStatus(obj)                    % Calls the solstis method 'status' and fills prefs in.
+            % Get status report from SolsTiS laser and update fields
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver) || isempty(obj.moduleName)
+                obj.etalon_lock = NaN;
+                obj.etalon_voltage = NaN;
+%                 obj.resonator_lock = false;
+                obj.resonator_voltage = NaN;
+                obj.output_monitor = 0;
+            else
+                reply = obj.com('status', 'solstis');
+                
+                obj.etalon_lock =       strcmp(reply.etalon_lock,'on');
+                obj.etalon_voltage =    reply.etalon_voltage;
+%                 obj.resonator_lock =    strcmp(reply.cavity_lock,'on');
+                obj.resonator_voltage = reply.resonator_voltage;
+                obj.output_monitor =    reply.output_monitor;
+            end
+            
+            obj.callGetWavelength()
+        end
+        
+        function getWavemeterWavelength(obj)        % Measures the output wavelength directly from the wavemeter. Also uses the EMM channel if the EMM module is active.
+            if isempty(obj.hwserver) || ~isvalid(obj.hwserver)
+                obj.NIR_wavelength =  NaN;
+                obj.VIS_wavelength =  NaN;
+                obj.diff_wavelength = 1950;
+            else
+                obj.NIR_wavelength = obj.hwserver.com('wavemeter', 'GetWavelengthNum', obj.NIR_channel, 0);
+
+                if strcmp(obj.determineModule(obj.setpoint_), obj.moduleVIS)
```

Comment:
I'm not familiar with this syntax; what does the underscore afterward do?

Reply:
Makes it different from the tunablelaser-required setpoint.